test/conftest.py

```diff
@@ -16,7 +16,7 @@ def help():
     """)
 
 
-def get_engine():
+def get_software_engine():
```

Comment:
Looking at the rest of the code would keeping get_engine() be more ideal? Instead of repeating code with get_software_engine() and get_cloud_engine(), you could have a get_engine(type) function instead, with a separate function get_environment_variable(variable_name). Then you could have:

```
def get_environment_variable(variable_name):
    if not variable_name in os.environ:
        help()
        return
    return os.environ[variable_name]

def get_engine(type):
    if type == "cloud":
        return create_engine(get_environment_variable("DREMIO_CLOUD_CONNECTION_URL"))
    elif type == "software":
        return create_engine(get_environment_variable("DREMIO_CONNECTION_URL"))
    else:
        print(f"Cannot create engine of type {type}")
```

Reply:
Another alternative is using inherited classes which in hindsight may be even more elegant:
```
def get_environment_variable(variable_name):
    if not variable_name in os.environ:
        help()
        return
    return os.environ[variable_name]

class engine_factory:
    connect_string = ""
    def get_engine(self):
        return create_engine(self.connect_string)

class software_engine_factory:
    def __init__(self):
         self.connect_string = get_environment_variable("DREMIO_CONNECTION_URL")

class cloud_engine_factory:
    def __init__(self):
         self.connect_string = get_environment_variable("DREMIO_CLOUD_CONNECTION_URL")
```

With these classes, to get a software engine, you just have to use 
`engine = software_engine_factory.get_engine()`